src/model/WinningLottoModel.js

```diff
@@ -0,0 +1,37 @@
+import Lotto from '../Lotto.js';
+import { RANK_NAME } from '../constant/system.js';
+import Validator from '../Validator.js';
+
+export default class WinningLottoModel extends Lotto {
+  #bonusNumber;
+
+  constructor(numbers) {
+    super(numbers);
+    this.#bonusNumber = null;
+  }
+
+  setBonusNumber(bonusNumber) {
+    Validator.bonusNumber(this.getNumbers(), bonusNumber);
+    this.#bonusNumber = bonusNumber;
+  }
+
+  calculateRank(lottoNumbers) {
+    const intersectionLottoNumbers = lottoNumbers.filter((number) =>
+      this.getNumbers().includes(number),
+    );
+
+    const matchCount = intersectionLottoNumbers.length;
```

Comment:
이 사이에 개행은 어떤 이유로 있는 건가요?

[이 부분은 개인적인 취향이니 참고만 해주세요 ㅎㅎ]
`matchCount`라는 변수에 `length`를 바로 넣어서 축약할 수 있을 것 같습니다.
그리고 `filter`내부 함수에서 `this.getNumbers()`를 변수화하여 가독성과 의미부여에서 이점을 얻을 수 있을 것 같습니다! 

```suggestion
    const matchCount = lottoNumbers.filter((number) => {
      const lottoNumbers = this.getNumbers();
      lottoNumbers.includes(number);
    }).length;
```

Reply:
개행을 준 이유는 딱히 없네요... 의미있게 사용하고자 했는데 놓친 부분이 많네요..! 

---
제가 이렇게 나눠 작성한 이유는 코드를 읽는 사람을 고려하여, 변수명을 사용해 조금 더 쉽게 이해할 수 있도록 했습니다!
지예님 방식도 정말 좋은 방법이라고 생각합니다!!!